source/event_stream.c

```diff
@@ -0,0 +1,1118 @@
+    aws_write_u64(header.header_value.static_val, (uint64_t)value);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int aws_event_stream_add_bytebuf_header(struct aws_array_list *headers, const char *name, uint8_t name_len,
+                                        uint8_t *value, uint16_t value_len, int8_t copy) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = value_len,
+            .value_owned = copy,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_BYTE_BUF
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+
+    if (copy) {
+        header.header_value.variable_len_val = aws_mem_acquire(headers->alloc, value_len);
+        if (!header.header_value.variable_len_val) {
+            return aws_raise_error(AWS_ERROR_OOM);
+        }
+
+        header.value_owned = 1;
+        memcpy((void *) header.header_value.variable_len_val, (void *) value, value_len);
+    } else {
+        header.value_owned = 0;
+        header.header_value.variable_len_val = (const uint8_t *) value;
+    }
+
+    if (aws_array_list_push_back(headers, (void *) &header)) {
+        aws_mem_release(headers->alloc, (void *) header.header_value.variable_len_val);
+        return AWS_OP_ERR;
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+int aws_event_stream_add_timestamp_header(struct aws_array_list *headers, const char *name, uint8_t name_len,
+                                          int64_t value) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = sizeof(uint64_t),
+            .value_owned = 0,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_TIMESTAMP,
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+    aws_write_u64(header.header_value.static_val, (uint64_t)value);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int aws_event_stream_add_uuid_header(struct aws_array_list *headers, const char *name, uint8_t name_len,
+                                     const uint8_t *value) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = 16,
+            .value_owned = 0,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_UUID,
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+    memcpy((void *) header.header_value.static_val, value, 16);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int8_t aws_event_stream_header_value_as_byte(struct aws_event_stream_header_value_pair *header) {
+    return (int8_t)header->header_value.static_val[0];
+
+}
+
+const char *aws_event_stream_header_value_as_string(struct aws_event_stream_header_value_pair *header) {
+    return (const char *) header->header_value.variable_len_val;
```

Comment:
I don't think you're currently writing the null terminator to the buffer. So this API is misleading because it doesn't return a valid C-string.

I've looked and you're passing strlen() as the length.

Reply:
this is no longer an issue since we just return a byte buffer.